### PR TITLE
NIAD-1023: Add SG6 segment code

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
## Description

Add two new fields for SG6:
* ReferenceServiceSubject (`RFF+SSI`)
* UnstructuredAddress (`ADR`)

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-1023

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [x] Acceptance Criteria met
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [x] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [x] Add/update any relevant documentation

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ReferenceServiceSubject.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * E.g. {@code}RFF+SSI:X88442211'{@code}
```
Comment (andrewwa-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ReferenceServiceSubject.java:13]: Fix javadoc
```suggestion
 * E.g. {@code RFF+SSI:X88442211'}
```

**Review by saileshp-kainos — DISMISSED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ReferenceServiceSubject.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+        // no op
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        if (StringUtils.isEmpty(number)) {
```
Comment (n-r-b) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ReferenceServiceSubject.java:56]: I would usually use `isBlank()` over `isEmpty()` as it also checks if a string contains just whitespace

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ReferenceServiceSubject.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * E.g. {@code RFF+SSI:X88442211'}
+ */
+@EqualsAndHashCode(callSuper = true)
+@Getter
+@Setter
```
Comment (n-r-b) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ReferenceServiceSubject.java:18]: is this used?

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/UnstructuredAddress.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * E.g. {@code ADR++US:FLAT1:12 BROWNBERRIE AVENUE::LEEDS:++LS18 5PN'} has no address parts 3 or 5.
+ * {@code ADR++++LS18 5PN'} has only a postcode
+ */
+@EqualsAndHashCode(callSuper = true)
+@Getter
+@Setter
```
Comment (n-r-b) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/UnstructuredAddress.java:17]: Is this used?

**Comment by andrewwa-kainos** [comment]
Will merge in the changes from #16 and make use of the exposed separator constants.

**Review by saileshp-kainos — APPROVED** [review]

**Review by n-r-b — APPROVED** [review]

**Pull request merged**